src/main/java/pl/piekoszek/gorskimatches/equation/SolutionToQuizzesMapper.java

```diff
@@ -52,4 +104,33 @@ public Map<String, Set<String>> insideSingleNumber(String solution) {
         }
         return quizzesAndSolutions;
     }
+
+    public Map<String, Set<String>> insideEquation(String solution) {
+        Map<String, Set<String>> quizzesAndSolutionsWithinEquation = new HashMap<>();
+
+        for (int i = 0; i < solution.length(); i++) {
+            char numberOrSymbolToBeReplaced = solution.charAt(i);
+            var numbersAndSymbolsToBeTakenFrom = takeOneMatchFromNumberOrSymbol.get(numberOrSymbolToBeReplaced);
```

Comment:
I'm not sure if it's `numbersAndSymbolsToBeTakenFrom`. I think it's something after taking.